src/lib.rs

```diff
@@ -189,3 +219,238 @@ impl From<Options> for original::Options {
+        matches!(
+            (self, other),
+            (HeadingLevel::H1, original::HeadingLevel::H1)
+                | (HeadingLevel::H2, original::HeadingLevel::H2)
+                | (HeadingLevel::H3, original::HeadingLevel::H3)
+                | (HeadingLevel::H4, original::HeadingLevel::H4)
+                | (HeadingLevel::H5, original::HeadingLevel::H5)
+                | (HeadingLevel::H6, original::HeadingLevel::H6)
+        )
+    }
+}
+
+impl From<Range> for std::ops::Range<usize> {
+    fn from(r: Range) -> Self {
+        std::ops::Range {
+            start: r.start.try_into().unwrap(),
+            end: r.end.try_into().unwrap(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::pulldown::Pulldown as _;
+
+    #[test]
+    fn convert_markdown_to_html_() {
+        let markdown_str = r#"
+hello
+=====
+
+* alpha
+* beta
+"#;
+
+        let result = Pulldown::markdown_to_html(markdown_str.to_string());
+
+        assert_eq!(
+            result,
+            r#"<h1>hello</h1>
+<ul>
+<li>alpha</li>
+<li>beta</li>
+</ul>
+"#
+        );
+    }
+
+    #[test]
+    fn parser_with_option_strikethrough() {
+        let markdown_input = "Hello world, this is a ~~complicated~~ *very simple* example.";
+
+        let option = Options::ENABLE_STRIKETHROUGH;
+
+        let iters: Vec<OffsetItem> =
+            Pulldown::parse_with_options(markdown_input.to_string(), option);
+
+        let expected: Vec<(original::Event, std::ops::Range<usize>)> =
+            original::Parser::new_ext(markdown_input, option.into())
+                .into_offset_iter()
+                .collect();
+
+        assert_eq!(iters.len(), expected.len());
+        for (a, b) in iters.into_iter().zip(expected) {
+            assert_eq!(a.0, b.0);
+            assert_eq!(a.1, b.1);
+        }
+    }
+    #[test]
+    fn html_test_1() {
+        let original = r##"Little header
+<script type="text/js">
+function some_func() {
+console.log("teeeest");
+}
+function another_func() {
+console.log("fooooo");
+}
+</script>"##;
+        let expected = r##"<p>Little header</p>
+<script type="text/js">
+function some_func() {
+console.log("teeeest");
+}
+function another_func() {
+console.log("fooooo");
+}
+</script>"##;
+
+        let result = Pulldown::markdown_to_html(original.to_string());
+        assert_eq!(result, expected);
+    }
+    #[test]
+    fn html_test_2() {
+        let original_string = r##"Little header
+<script
+type="text/js">
+function some_func() {
+console.log("teeeest");
+}
+function another_func() {
+console.log("fooooo");
+}
+</script>"##;
+        let expected = r##"<p>Little header</p>
+<script
+type="text/js">
+function some_func() {
+console.log("teeeest");
+}
+function another_func() {
+console.log("fooooo");
+}
+</script>"##;
+        let result = Pulldown::markdown_to_html(original_string.to_string());
+        assert_eq!(result, expected);
+    }
+}
```

Comment:
We probably only need 1 HTML test. The goal is to see whether we are calling `pulldown-cmark` correctly, rather than making sure `pulldown-cmark` can handle all sorts of markdown strings.